src/mixins/README.md

```diff
@@ -0,0 +1,36 @@
+# Mixins
+
+A Mixin is a class which takes the input of a base class and adds / modifies existing functions / static values and returns a new class which can be instantiated. The type of returned class will still be the base class.
+
+Example usage of Mixin is as follows: 
+
+`const newClass = MixinClass(BaseClass)`
+
+Now NewClass will have functions of MixinClass & BaseClass
+
+An important note is that MixinClasses can be combined! This allows us to write small & simple Mixins that a user can compose into a powerful class as needed. We can also write a helper function to mixin a whole bunch of smaller mixins. Again this will allow us to write simpler Mixins & compose them to be more powerful, mixins. 
```

Comment:
I'd drop the _written_ example on Mixins and instead just describe how we leverage them for extensions.
A code example is fine since the _concept_ of a mixin and the many ways to achieve it are different between languages.
I'd also move up the explanation on what Mixins are in TypeScript so that confused users can quickly find the resource they need to fill their knowledge gap and come back.